### PR TITLE
feat(dynamic-flows): Update the array schema's "required" behaviour

### PR DESCRIPTION
**Pull request by dhananjaykha**
## 🖼 Context

EARLIER BEHAVIOUR:
- For a required array schema input, an empty array is NOT a valid input
- The array must always contain at least one element.

UPDATE:
- For a required array schema input, an empty array is a valid input
- For the at least one restriction the `minItems` construct must be used.

## 🚀 Changes

 
- Removes the check for required array to have at least one element
- Fix: an invocation to the utility function not passing the `required` value to the function.
- removes test expecting the earlier behaviour

## ✅ Checklist

- [x] Make PR title meaningful and follow [the commit lint format](https://github.com/transferwise/neptune-web/blob/master/CONTRIBUTING.md#versioning-and-commit-lint) (it will appear in the changelog)
- [x] Changes are tested and all tests pass
- [x] Changes meet [accessibility standards](https://github.com/transferwise/neptune-web/blob/main/ACCESSIBILITY.md) and there are no violations in the console
- [ ] Changes work in all supported browsers (don't forget IE11)
- [ ] You've updated the documentation if necessary

### CODE REVIEW

File: packages/dynamic-flows/src/common/validation/rule-validators/index.js [hunk 1]
```diff
@@ -4,9 +4,8 @@ function isValidRequired(value, isRequired) {
   return !isRequired || !isUndefined(value);
 }
 
-function isValidRequiredArray(value, isRequired) {
-  const isNonEmptyArray = isArray(value) && value.length > 0;
-  return !isRequired || isNonEmptyArray;
+function isValidRequiredArray(value, isRequired = false) {
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/common/validation/rule-validators/index.js:7]: is there a reason we need to default to `false` here?

- the `isValidRequired` on line 3 doesn't have a default.
- all the functions in validation-failures/index.js don't have a default either.

File: packages/dynamic-flows/src/common/validation/validation-failures/index.js [hunk 2]
```diff
@@ -141,7 +141,7 @@ function getConstValidationFailures(value, schema, isRequired) {
   return [];
 }
 
-function getArrayValidationFailures(value, schema, isRequired) {
+function getArrayValidationFailures(value, schema, isRequired = false) {
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/common/validation/validation-failures/index.js:144]: is there a reason for having a default value while all the other functions in this module don't?

File: packages/dynamic-flows/src/common/validation/rule-validators/index.js (hunk 1 above)
Comment (dhananjaykha) [reply packages/dynamic-flows/src/common/validation/rule-validators/index.js:7]: Sure. I felt the default makes it predictable that all falsy values are defaulted to false. But okay to make it consistent.

File: packages/dynamic-flows/src/common/validation/validation-failures/index.js (hunk 2 above)
Comment (dhananjaykha) [reply packages/dynamic-flows/src/common/validation/validation-failures/index.js:144]: Only reason to add it was if some usage does not pass it explicitly, it is mapped to false. Will make it consistent. No hard opinion.

File: packages/dynamic-flows/src/common/validation/rule-validators/spec.js [hunk 3]
```diff
@@ -141,4 +142,24 @@ describe('Given a library for validating json schema rules', () => {
...
     });
   });
+
+  describe('when establishing validations an array', () => {
+    it('should return true all values for a non-required input', () => {
+      expect(isValidRequiredArray([1])).toBe(true);
+      expect(isValidRequiredArray([])).toBe(true);
+
+      expect(isValidRequiredArray(undefined)).toBe(true);
+      expect(isValidRequiredArray(null)).toBe(true);
```
Comment (JulianG-TW) [inline packages/dynamic-flows/src/common/validation/rule-validators/spec.js:152]: missing a second argument in these calls?

**Review by JulianG-TW — APPROVED** [review]

**Pull request merged**